tests/difference/core/quint_model/ccv.qnt

```diff
@@ -0,0 +1,877 @@
+    pure def GetEmptyConsumerState(): ConsumerState = 
+        {
+            chainState: GetEmptyChainState,
+            maturationTimes: Map(),
+            outstandingPacketsToProvider: List(),
+            receivedVSCPackets: List(),
+        }
+
+    // the state of the protocol consists of the composition of the state of one provider chain with potentially many consumer chains.
+    type ProtocolState = {
+        providerState: ProviderState,
+        // the state of each consumer chain.
+        // note that we assume that this contains all consumer chains that may ever exist,
+        // and consumer chains that are currently not running will have providerState.consumerStatus == UNUSED or STOPPED.
+        consumerStates: Chain -> ConsumerState
+    }
+
+    // gets a protocol state that is initialized minimally.
+    pure def GetEmptyProtocolState(): ProtocolState = 
+        {
+            providerState: GetEmptyProviderState,
+            consumerStates: Map(),
+        }
+
+    type Error = {
+        message: str
+    }
+
+    // we return either a result or an error.
+    // if hasError is true, newState may be arbitrary, but the error will be meaningful.
+    // if hasError is false, error may be arbitrary, but newState will be meaningful.
+    type Result = {
+        hasError: bool,
+        newState: ProtocolState,
+        error: Error
+    }
+
+    pure def Ok(newState: ProtocolState): Result = {
+        {
+            hasError: false,
+            newState: newState,
+            error: {
+                message: ""
+            }
+        }
+    }
+
+    pure def Err(msg: str): Result = {
+        {
+            hasError: true,
+            newState: {
+                providerState: {
+                    chainState: {
+                        votingPowerHistory: List(),
+                        currentValidatorSet: Map(),
+                        lastTimestamp: 0,
+                    },
+                    outstandingPacketsToConsumer: Map(),
+                    receivedMaturations: Set(),
+                    sentVSCPackets: Map(),
+                    providerValidatorSetChangedInThisBlock: false,
+                    consumerStatus: Map(),
+                    runningVscId: 0,
+                },
+                consumerStates: Map(),
+            },
+            error: {
+                message: msg
+            }
+        }
+    }
+
+    // possible consumer statuses
+    pure val STOPPED = "stopped" // the chain was once a consumer chain, but has been voluntarily dropped by the provider.
+    pure val TIMEDOUT = "timedout" // the chain has timed out and was dropped by the provider. This is only used for involuntary drops.
+    pure val RUNNING = "running" // the chain is currently a consumer chain. Running chains are those that get sent VSCPackets.
+    pure val UNUSED = "unused" // the chain has never been a consumer chain, and is available to become one.
```

Comment:
nit: Not sure how I feel about the naming, especially for `UNUSED`, in the sense that consumer chains already run.  Although, I see that the [UNUSED](https://github.com/cosmos/ibc/pull/911/files#diff-ed9c0aa72ff4479e19c87de5d812a7e9c41151d0a73f908070930145bd916b4fR130) stems from the TLA+ spec. I cannot think of any succinct name but would it make sense to have something like `NOT_CONSUMNG_SECURITY`, etc.

Reply:
Let me comment on this:
> in the sense that consumer chains already run.
This part is not true - UNUSED consumers might already run, or they might not. 

But you are right that the current name is not great. I'll call it `NOT_CONSUMER`